sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/ZPageHttpHandler.java

```diff
@@ -75,9 +76,24 @@
   @Override
   public final void handle(HttpExchange httpExchange) throws IOException {
     try {
+      String requestMethod = httpExchange.getRequestMethod();
       httpExchange.sendResponseHeaders(200, 0);
-      zpageHandler.emitHtml(
-          parseQueryMap(httpExchange.getRequestURI()), httpExchange.getResponseBody());
+      if (requestMethod.equalsIgnoreCase("GET")) {
+        zpageHandler.emitHtml(
+            requestMethod,
+            parseQueryString(httpExchange.getRequestURI().getRawQuery()),
+            httpExchange.getResponseBody());
+      } else if (requestMethod.equalsIgnoreCase("POST")) {
+        // BufferedReader requestBodyReader =
+        // new BufferedReader(new InputStreamReader(httpExchange.getRequestBody(), "utf-8"));
+        // String queryString = requestBodyReader.readLine();
+        InputStreamReader requestBodyReader =
+            new InputStreamReader(httpExchange.getRequestBody(), "utf-8");
+        String queryString = CharStreams.toString(requestBodyReader);
+        requestBodyReader.close();
+        zpageHandler.emitHtml(
```

Comment:
rather than just having a single `emitHtml` method, it would be better to have 2 methods, I think.  One to process the POST changes, and one to then emit the resulting HTML page. What do you think about splitting the interface in 2, and then the things that don't support any updates can just no-op the implementation on the `processUpdate` method (or whatever you call it)?

Reply:
So one for normally emitting HTML on GET, and another one for processing updates on POST (or any other request methods) then?

Reply:
right, and if a POST is received, first you call "updateChanges", then you call "emitHtml". If a GET is received, you just call "emitHtml" and you're done.